exporter/signalfxexporter/internal/translation/dpfilters/propertyfilter.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dpfilters // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter/internal/translation/dpfilters"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/confmap"
+)
+
+var anyREFilter = func() *StringFilter {
+	if sf, err := NewStringFilter([]string{"/^.*$/"}); err != nil {
```

Comment:
I would expect `NewStringFilter(nil)` to have the same effect... Do you have know why it's not the case by a chance? If it had such behavior, we wouldn't need the custom unmarshaller on `PropertyFilter`

Reply:
I'm not sure I agree that I'd expect that to be the case since `NewStringFilter(nil)` to me suggests a new string filter that matches nothing. Added a unit test to confirm this is the existing behavior.

I just tried adding a nil pointer `Matches()` true default to make a custom PropertyFilter unmarshaller unneeded since it is a little wonky as is. I'm not super confident it won't change existing behavior though.